### PR TITLE
replace http-errors with @fastify/error

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/on-error.js [hunk 1]
```diff
@@ -34,8 +34,8 @@ async function main () {
     reply.from(`http://localhost:${target.server.address().port}/`,
       {
         onError: (reply, { error }) => {
-          t.equal(error.status, 504)
-          reply.code(error.status).send(error)
+          t.equal(error.statusCode, 504)
```
Comment (Uzlopak) [inline test/on-error.js:37]: I assume because I had to change this unit test, it will be a breaking change

File: index.js [hunk 2]
```diff
@@ -137,11 +143,15 @@ module.exports = fp(function from (fastify, opts, next) {
         this.request.log.warn(err, 'response errored')
         if (!this.sent) {
           if (err.code === 'ERR_HTTP2_STREAM_CANCEL' || err.code === 'ENOTFOUND') {
-            onError(this, { error: new createError.ServiceUnavailable() })
+            onError(this, { error: ServiceUnavailableError() })
           } else if (err instanceof TimeoutError || err.code === 'UND_ERR_HEADERS_TIMEOUT') {
-            onError(this, { error: new createError.GatewayTimeout() })
+            onError(this, { error: new GatewayTimeoutError() })
+          } else if (err.code === 'ECONNRESET') {
+            onError(this, { error: new ConnectionResetError() })
```
Comment (Uzlopak) [inline index.js:150]: I created these special cases. Does it make sense?

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/errors.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const createError = require('@fastify/error')
```
Comment (mcollina) [inline lib/errors.js:1]:
```suggestion
'use strict'

const createError = require('@fastify/error')
```

File: index.js (hunk 2 above)
Comment (mcollina) [reply index.js:150]: yes, that's ok

File: test/on-error.js (hunk 1 above)
Comment (Eomm) [reply test/on-error.js:37]: Is the error a `GatewayTimeoutError` instance?

Comment (Uzlopak) [reply test/on-error.js:37]: I will extend the test to check if it is a specific error.

Comment (Uzlopak) [reply test/on-error.js]: Changed it to check for the Error. Does this help?

**Comment by mcollina** [comment]
@Eomm PTAL

**Review by Eomm — APPROVED** [review]

**Pull request merged**